L1Trigger/Phase2L1ParticleFlow/plugins/L1NNTauProducer.cc

```diff
@@ -0,0 +1,112 @@
+#include "L1Trigger/Phase2L1ParticleFlow/interface/L1NNTauProducer.hh"
+#include <TLorentzVector.h>
+#include <cmath>
+
+L1NNTauProducer::L1NNTauProducer(const edm::ParameterSet& cfg)
+    : fSeedPt_(cfg.getParameter<double>("seedpt")),
+      fConeSize_(cfg.getParameter<double>("conesize")),
+      fTauSize_(cfg.getParameter<double>("tausize")),
+      fMaxTaus_(cfg.getParameter<int>("maxtaus")),
+      fNParticles_(cfg.getParameter<int>("nparticles")),
+      fL1PFToken_(consumes<vector<l1t::PFCandidate> >(cfg.getParameter<edm::InputTag>("L1PFObjects"))) {
+  std::string lNNFile = cfg.getParameter<std::string>("NNFileName");  //,"L1Trigger/Phase2L1Taus/data/tau_3layer.pb");
+  fTauNNId_ = std::make_unique<TauNNId>();
+  if (lNNFile.find("v0") == std::string::npos)
+    fTauNNId_->initialize("input_1:0", lNNFile, fNParticles_);
+  else if (lNNFile.find("v0") != std::string::npos)
+    fTauNNId_->initialize("dense_1_input:0", lNNFile, fNParticles_);
```

Comment:
with the removal of delayed initialization as mentioned above, these five lines can become:
```cpp
fTauNNId_ = std::make_unique<TauNNId>(lNNFile.find("v0") == std::string::npos ? "input_1:0" : "dense_1_input:0", lNNFile, fNParticles_);
```

Reply:
done